api/v1alpha1/hardwareclassification_types.go

```diff
@@ -27,111 +27,135 @@ type HardwareClassificationSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// HardwareCharacteristics defines expected hardware configurations for Cpu, Disk, Nic and Ram.
-	HardwareCharacteristics HardwareCharacteristics `json:"hardwareCharacteristics,omitemty"`
+	// ExpectedHardwareConfiguration defines expected hardware configurations for CPU, RAM, Disk, NIC.
+	ExpectedHardwareConfiguration ExpectedHardwareConfiguration `json:"expectedValidationConfiguration"`
 }
 
 // ExpectedHardwareConfiguration details to match with the host
-type HardwareCharacteristics struct {
+type ExpectedHardwareConfiguration struct {
 	// +optional
-	Cpu *Cpu `json:"cpu,omitempty"`
```

Comment:
Don't remove this, As per K8s standard, we named it like that.

Reply:
Changed back.